examples/java-matter-controller/java/src/com/matter/controller/commands/pairing/PairingCommand.java

```diff
@@ -47,6 +48,97 @@ public abstract class PairingCommand extends MatterCommand {
   private final StringBuffer mOnboardingPayload = new StringBuffer();
   private final StringBuffer mDiscoveryFilterInstanceName = new StringBuffer();
 
+  public long getNodeId() {
+    return mNodeId.get();
+  }
+
+  public int getSetupPINCode() {
+    return mSetupPINCode.get();
+  }
+
+  public int getDiscriminator() {
+    return mDiscriminator.get();
+  }
+
+  public long getTimeoutMillis() {
+    return mTimeoutMillis.get();
+  }
+
+  @Override
+  public void onConnectDeviceComplete() {
+    System.out.println("onConnectDeviceComplete");
+  }
+
+  @Override
+  public void onStatusUpdate(int status) {
+    System.out.println("onStatusUpdate with status: " + status);
+  }
+
+  @Override
+  public void onPairingComplete(int errorCode) {
+    System.out.println("onPairingComplete with error code: " + errorCode);
+    if (errorCode != 0) {
+      setTestResult("Failure");
+      throw new RuntimeException("onPairingComplete");
+    }
+  }
+
+  @Override
+  public void onPairingDeleted(int errorCode) {
+    System.out.println("onPairingDeleted with error code: " + errorCode);
+  }
+
+  @Override
+  public void onCommissioningComplete(long nodeId, int errorCode) {
+    System.out.println("onCommissioningComplete with error code: " + errorCode);
+    if (errorCode == 0) {
+      setTestResult("Success");
+    } else {
+      setTestResult("Failure");
+      throw new RuntimeException("onCommissioningComplete");
+    }
+  }
+
+  @Override
+  public void onReadCommissioningInfo(
+      int vendorId, int productId, int wifiEndpointId, int threadEndpointId) {
+    System.out.println("onReadCommissioningInfo");

```

Comment:
should we use java.util.logging infrastructure instead?

Reply:
I see there are significant amount of places that use system log, I would update them in one dedicate PR separately, thanks for the suggestion.
https://github.com/project-chip/connectedhomeip/issues/23731